src/com/hapgood/refreshable.cljc

```diff
@@ -77,49 +119,61 @@
-                                control (when (not (nil? event))
-                                          (if alarm
-                                            [(async/timeout 0) source failsafe called-at backoffs']
-                                            [alarm source failsafe called-at backoffs'])))]
-          (recur a s f c bs)
+        (if-let [[a s f c bs o] (condp = port
+                                  alarm (let [source (async/chan 1)]
+                                          (try (acquire source)
+                                               [nil source (when failsafe-timeout (async/timeout failsafe-timeout)) now backoffs' old]
+                                               (catch #?(:clj java.lang.Exception :cljs js/Error) e
+                                                 (when-let [backoff (error-handler refreshable {:error-type ::exception :exception e :retry (first backoffs')})]
+                                                   [(async/timeout backoff) nil nil nil (rest backoffs') old]))))
+                                  source (let [validator (get-validator refreshable)]
+                                           (cond
+                                             (nil? event)
+                                             (when-let [backoff (error-handler refreshable {:error-type ::source-closed :retry (first backoffs')})]
+                                               [(async/timeout backoff) nil nil nil (rest backoffs') old])
+
+                                             (not (valid? validator event))
+                                             (when-let [backoff (error-handler refreshable {:error-type ::validation-error :retry (first backoffs')})]
+                                               [(async/timeout backoff) nil nil nil (rest backoffs') old])
+
+                                             :else
+                                             (let [latency (delta-t called-at now)
+                                                   refresh-after (max 0 (- interval latency))]
+                                               (vary-meta refreshable #(-> %
+                                                                           (merge {::acquired-at now ::latency latency})
+                                                                           (dissoc ::error)
+                                                                           (update ::version inc)))
+                                               (let [[pc pc'] (reset-vals! out-ref (async/promise-chan))]
+                                                 (async/offer! pc event) ; release any previously blocked takes
+                                                 (async/offer! pc' event))
+                                               (notify-watches* refreshable watches old event)
+                                               [(async/timeout refresh-after) nil nil nil backoffs event])))
+
```

Comment:
Blank line here seems inconsistent.... which is an argument for not having them anywhere in a top level form!